### PR TITLE
First version

### PR DESCRIPTION
**Pull request by dawnerd**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
+const soap = require('soap');
+const Joi = require('joi');
+const { promisify: prom } = require('util');
+
+let SoapClient;
+let url = 'https://devtest.dotfit.com/webservices/OrdersService.asmx?WSDL';
```
Comment (jgallen23) [inline index.js]: we need to be able to change this based on dev/prod

Comment (dawnerd) [reply index.js]: https://github.com/firstandthird/node-dotfit/pull/1/files#diff-1e758beff86240736d07379a17316a7bR7

File: index.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+module.exports = {
```
Comment (jgallen23) [inline index.js]: any reason not to go with a class - especially since you have gets and sets

File: index.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    if (process.env.DEBUG) {
+      // eslint-disable-next-line no-console
+      console.log(tags, msg);
+    }
+  },
+  async createClient() {
+    this.log(['info'], `Creating client for WSDL: ${url}`);
+    const client = await soap.createClientAsync(url);
+
+    const ClubID = process.env.CLUB_ID;
```
Comment (jgallen23) [inline index.js]: I don't really like libraries reading from env vars. can we pass in?

File: index.js (hunk 2 above)
Comment (jgallen23) [reply index.js]: I would think we'd do new `new DotFitApi({ creds, env: 'dev|prod' )`

File: index.js [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+      };
+    }
+
+    return response.GetShipmentInfoByIDResult.ShippedOrder;
+  },
+  async createOrder(orderData) {
+    const client = await this.client;
+
+    const schema = Joi.object().keys({
+      wholesaleUserID: Joi.number().optional().default(process.env.WHOLESALE_ID),
```
Comment (jgallen23) [inline index.js]: could we make wholesale_id passed in when instantiating?

File: index.js (hunk 2 above)
Comment (dawnerd) [reply index.js]: I had a class originally but I'll switch it out and see how it does

File: index.js (hunk 4 above)
Comment (jgallen23) [reply index.js]: ping on this - I don't think env vars should be used in libraries

Comment (dawnerd) [reply index.js]: Hah, thought I got that one last pass, guess not. Fixed!

**Pull request merged**